src/containers/BooksForm.js

```diff
@@ -1,27 +1,64 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 import { categories } from '../constant';
+import { createBook } from '../actions/index';
 
-const BooksForm = () => (
-  <form>
-    <div className="form-group">
-      <input type="text" className="form-control" id="title" placeholder="Title" />
-    </div>
-    <div className="form-group">
-      <select className="form-control" id="category">
-        {
+const mapDispatchToProps = dispatch => ({
+  createBook: book => dispatch(createBook(book)),
+});
+
+const BooksForm = ({
+  createBook,
+}) => {
+  const intialstate = {
+    title: '',
+    category: categories[0],
+  };
+
+  const [state, setstate] = useState(intialstate);
+
+  const handleChange = event => {
+    event.preventDefault();
+    setstate({
+      ...state,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = () => {
+    if (state.title === '') return;
+    createBook({
+      ...state,
+      id: Math.floor(Math.random() * 100),
```

Comment:
- [ ] Depending in `random` to generate unique ids doesn't guarantee uniqueness. I got this console error when I tried to add a new book which happens to generate the same id for another book.

![Screenshot from 2021-01-16 17-03-01](https://user-images.githubusercontent.com/4246637/104815372-be4d5600-581c-11eb-810a-d8c320da605b.png)

I suggest that you depend on another technique like `Date.now()` which will always guarantee a unique ID or use [uuid](https://www.npmjs.com/package/uuidv4) library.